Compile bug: Build fails on ppc64le

### Git commit


commit 5f5e39e1ba5dbea814e41f2a15e035d749a520bc (HEAD -> master, tag: b5215, origin/master, origin/HEAD)

### Operating systems

Mac, Linux

### GGML backends

CPU

### Problem description & steps to reproduce

While trying to build llama.cpp on ppc64le, build fails with compilation errors. 

### First Bad Commit

commit c6e8cc28c15166dba15629dba6a7366d4d5955ca
Author: Acly <aclysia@gmail.com>
Date:   Thu Apr 17 14:16:45 2025 +0200

    ggml : Depthwise 2D convolution (ggml/1152)
    
    * ggml-cpu : kernels for faster depthwise 2D convolution
    
    * fix compile: remove static after moving to ops.cpp
    
    * add dilation for depthwise_conv_2d
    
    * review: rename to ggml_conv_2d_dw_direct, remove redundant struct keywords, pass by ref, whitespace
    
    * review: rename depthwise_conv_2d -> conv_2d_dw everywhere



### Compile command

```shell
cmake -B <build_dir> -DLLAMA_CURL=OFF
cmake --build <build_dir>
```

### Relevant log output

```shell
lama.cpp/ggml/src/ggml-cpu/simd-mappings.h:344:33: error: cannot convert ‘float’ to ‘__vector float’ {aka ‘__vector(4) float’} in initialization
  344 | #define GGML_F32x4_ZERO         0.0f
      |                                 ^~~~
      |                                 |
      |                                 float
llama.cpp/ggml/src/ggml-cpu/simd-mappings.h:372:29: note: in expansion of macro ‘GGML_F32x4_ZERO’
  372 | #define GGML_F32_VEC_ZERO   GGML_F32x4_ZERO
      |                             ^~~~~~~~~~~~~~~
llama.cpp/ggml/src/ggml-cpu/ops.cpp:6120:36: note: in expansion of macro ‘GGML_F32_VEC_ZERO’
 6120 |                 GGML_F32_VEC sum = GGML_F32_VEC_ZERO;
      |                                    ^~~~~~~~~~~~~~~~~
gmake[2]: *** [ggml/src/CMakeFiles/ggml-cpu.dir/build.make:233: ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/ops.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:2154: ggml/src/CMakeFiles/ggml-cpu.dir/all] Error 2
```

--- a/ggml/src/ggml-cpu/simd-mappings.h
+++ b/ggml/src/ggml-cpu/simd-mappings.h
@@ -341,7 +341,7 @@ static inline void __avx_f32cx8_store(ggml_fp16_t *x, __m256 y) {
 #define GGML_F32_EPR  4
 
 #define GGML_F32x4              vector float
-#define GGML_F32x4_ZERO         0.0f
+#define GGML_F32x4_ZERO         {0.0f}
 #define GGML_F32x4_SET1         vec_splats
 #define GGML_F32x4_LOAD(p)      vec_xl(0, p)
 #define GGML_F32x4_STORE(p, r)  vec_xst(r, 0, p)

